### ISSUE TITLE
Brackets around type operator :*: are removed

### ISSUE THREAD

**Issue by koterpillar** [ndmitchell/hlint#361]
```haskell
{-# LANGUAGE TypeOperators #-}
import Control.Applicative

import GHC.Generics

value :: [(Maybe :*: Maybe) Int]
value = liftA2 (:*:) [Just 1] [Just 2]
```

This is valid (nonsensical) module, but HLint removes brackets around `:*:`, making it a type error.

**Comment by koterpillar** [comment]
#227 looks related, but talks specifically about multiplication, so I'm not sure it's a duplicate.

**Comment by koterpillar** [comment]
Ignore this, it should be against hindent.
